### ISSUE TITLE
Type 'ShadowJar' property 'transformers.$0.serviceEntries' is missing an input or output annotation

### ISSUE THREAD

**Issue by effad** [johnrengelman/shadow#669]
When adding `mergeServiceFiles()` to my build setup the build fails. Without the `mergeServiceFiles()` it works.
I've tried to create a minimal example that demonstrates the problem but there it worked in both cases.

This is an excerpt from my build configuration:
```
		project.shadowJar {
			archiveClassifier.set('shadow')
			zip64 true
//			mergeServiceFiles()
//			into("Resources") {
//				from("src/dist");
//			}
//			transform(new ManifestMergeTransformer(excludePackages:  
//				"org/bouncycastle.*|" + 
//				"META-INF/versions/.*/org/bouncycastle.*"
//			))			
//			manifest {
//				attributes["Multi-Release"] = true
//			}
//			append 'META-INF/cxf/bus-extensions.txt'
//			append 'META-INF/mailcap'
		}
```

As soon as I enable the mergeServiceFiles() line, the build fails.

### Shadow Version
7.0.0

### Gradle Version
7.0

### Expected Behavior
Build should work

### Actual Behavior
Build fails with this error:
```
> Task :project:products:ArchiveServer:shadowJar FAILED
Could not create archive: /home/rli/devel/repository/source/java/project/products/ArchiveServer/build/libs/ArchiveServer-20.9.9.20-shadow.jar
org.gradle.internal.execution.WorkValidationException: A problem was found with the configuration of task ':project:products:ArchiveServer:shadowJar' (type 'ShadowJar').
  - Type 'ShadowJar' property 'transformers.$0.serviceEntries' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
```

**Comment by effad** [comment]
To workaround the problem I've now taken the Transformers and copied them (almost) verbatim into classes within our build system. The build.configuration now looks like this:
```
		project.shadowJar {
			archiveClassifier.set('shadow')
			zip64 true
//			mergeServiceFiles() 			
			transform(new SynShadowServiceFileTransformer());  
			into("Resources") {
				from("src/dist");
			}
			transform(new ManifestMergeTransformer(excludePackages:  
				"org/bouncycastle.*|" + 
				"META-INF/versions/.*/org/bouncycastle.*"
			))			
			manifest {
				attributes["Multi-Release"] = true
			}
			
//			append 'META-INF/cxf/bus-extensions.txt'
//			append 'META-INF/mailcap'
			transform(new SynShadowAppendingTransformer(resource: 'META-INF/cxf/bus-extensions.txt'));  
			transform(new SynShadowAppendingTransformer(resource: 'META-INF/mailcap'));  
		}
```
and without any further changes the problem is gone?! Something very strange seems to be going on in gradle...

**Comment by jrfaller** [comment]
I also have this issue and it's kind of blocking for me, but I am not sure to understand how you got it working @effad ? Could you point me to the resolving commit? Cheers.

**Comment by effad** [comment]
> I also have this issue and it's kind of blocking for me, but I am not sure to understand how you got it working @effad ? Could you point me to the resolving commit? Cheers.

As mentioned above, I've copied the files into our own build system (i.e. the extensions we've made for gradle) and -- to my surprise -- the problem went away.

So there's no commit to be shared :-(.

**Comment by johnrengelman** [comment]
Are you sure you are using Gradle 7? This seems to have been fixed here - https://github.com/johnrengelman/shadow/commit/8bbece8c7c20fd7123d81f8186eec6fd802fd73c#diff-895754e50811128b82057e1b75fbcb430fd044aca0bfc61661e4c24e5c6e6c4d which was included in v6.0.0 and beyond.

**Comment by nilols** [comment]
I had the same error, I was using Gradle 7.1.1 but the issue was a to old shadow plugin version, after bumping the shadow plugin version everything worked as expected again.
